Update AbstractChecks to log DetailAST - IllegalCatch

Child of https://github.com/checkstyle/checkstyle/issues/5777

This issue specifically focuses on IllegalCatch.

Can I get help with this? I'm getting this while doing ```mvn clean verify```
```
[ERROR] Failures:
[ERROR]   XpathRegressionIllegalCatchTest.testOne:59->AbstractXpathTestSupport.runVerifications:209->AbstractXpathTestSupport.verifyXpathQueries:125 Generated queries do not match expected ones ==>
expected: <[/CLASS_DEF[./IDENT[@text='SupressionXpathRegressionIllegalCatchOne']]/OBJBLOCK/METHOD_DEF[./IDENT[@text='foo']]/SLIST/LITERAL_TRY/LITERAL_CATCH]> 
but was: <[]>
```
The code is:
```
package org.checkstyle.suppressionxpathfilter.illegalcatch;

public class SuppressionXpathRegressionIllegalCatchOne {
	public void foo() {
		try {
		} catch (Exception e) { // warn
		}
	}
}
```

The AST is:
```
PACKAGE_DEF -> package [1:0]
|--ANNOTATIONS -> ANNOTATIONS [1:45]
|--DOT -> . [1:45]
|   |--DOT -> . [1:22]
|   |   |--DOT -> . [1:11]
|   |   |   |--IDENT -> org [1:8]
|   |   |   `--IDENT -> checkstyle [1:12]
|   |   `--IDENT -> suppressionxpathfilter [1:23]
|   `--IDENT -> illegalcatch [1:46]
`--SEMI -> ; [1:58]
CLASS_DEF -> CLASS_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|   `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_CLASS -> class [3:7]
|--IDENT -> SuppressionXpathRegressionIllegalCatchOne [3:13]
`--OBJBLOCK -> OBJBLOCK [3:55]
    |--LCURLY -> { [3:55]
    |--METHOD_DEF -> METHOD_DEF [4:1]
    |   |--MODIFIERS -> MODIFIERS [4:1]
    |   |   `--LITERAL_PUBLIC -> public [4:1]
    |   |--TYPE -> TYPE [4:8]
    |   |   `--LITERAL_VOID -> void [4:8]
    |   |--IDENT -> foo [4:13]
    |   |--LPAREN -> ( [4:16]
    |   |--PARAMETERS -> PARAMETERS [4:17]
    |   |--RPAREN -> ) [4:17]
    |   `--SLIST -> { [4:19]
    |       |--LITERAL_TRY -> try [5:2]
    |       |   |--SLIST -> { [5:6]
    |       |   |   `--RCURLY -> } [6:2]
    |       |   `--LITERAL_CATCH -> catch [6:4]
    |       |       |--LPAREN -> ( [6:10]
    |       |       |--PARAMETER_DEF -> PARAMETER_DEF [6:11]
    |       |       |   |--MODIFIERS -> MODIFIERS [6:11]
    |       |       |   |--TYPE -> TYPE [6:11]
    |       |       |   |   `--IDENT -> Exception [6:11]
    |       |       |   `--IDENT -> e [6:21]
    |       |       |--RPAREN -> ) [6:22]
    |       |       `--SLIST -> { [6:24]
    |       |           `--RCURLY -> } [7:2]
    |       `--RCURLY -> } [8:1]
    `--RCURLY -> } [9:0]


```

--- a/src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionIllegalCatchTest.java
+++ b/src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionIllegalCatchTest.java
@@ -1,0 +1,85 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.IllegalCatchCheck;
+
+public class XpathRegressionIllegalCatchTest extends AbstractXpathTestSupport {
+
+    private final String checkName = IllegalCatchCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionIllegalCatchOne.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(IllegalCatchCheck.class);
+
+        final String[] expectedViolation = {
+            "6:11: " + getCheckMessage(IllegalCatchCheck.class,
+                IllegalCatchCheck.MSG_KEY, "RuntimeException"),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+            "/CLASS_DEF[./IDENT[@text='SuppressionXpathRegressionIllegalCatchOne']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='fun']]/SLIST"
+                + "/LITERAL_TRY/LITERAL_CATCH"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+
+    @Test
+    public void testTwo() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionIllegalCatchTwo.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(IllegalCatchCheck.class);
+
+        final String[] expectedViolation = {
+            "16:11: " + getCheckMessage(IllegalCatchCheck.class,
+                IllegalCatchCheck.MSG_KEY, "java.lang.RuntimeException"),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+            "/CLASS_DEF[./IDENT[@text='SuppressionXpathRegressionIllegalCatchTwo']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='methodTwo']]/SLIST"
+                + "/LITERAL_TRY/LITERAL_CATCH"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+}

--- a/src/it/resources/org/checkstyle/suppressionxpathfilter/illegalcatch/SuppressionXpathRegressionIllegalCatchOne.java
+++ b/src/it/resources/org/checkstyle/suppressionxpathfilter/illegalcatch/SuppressionXpathRegressionIllegalCatchOne.java
@@ -1,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.illegalcatch;
+
+public class SuppressionXpathRegressionIllegalCatchOne {
+    public void fun() {
+        try {
+        } catch (RuntimeException e) { // warn
+        }
+    }
+}

--- a/src/it/resources/org/checkstyle/suppressionxpathfilter/illegalcatch/SuppressionXpathRegressionIllegalCatchTwo.java
+++ b/src/it/resources/org/checkstyle/suppressionxpathfilter/illegalcatch/SuppressionXpathRegressionIllegalCatchTwo.java
@@ -1,0 +1,19 @@
+package org.checkstyle.suppressionxpathfilter.illegalcatch;
+
+public class SuppressionXpathRegressionIllegalCatchTwo {
+    public void methodOne() {
+        try {
+            foo();
+        } catch (java.sql.SQLException e) {
+        }
+    }
+
+    private void foo() throws java.sql.SQLException {
+    }
+
+    public void methodTwo() {
+        try {
+        } catch (java.lang.RuntimeException e) { // warn
+        }
+    }
+}

--- a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java
@@ -62,9 +62,6 @@ import com.puppycrawl.tools.checkstyle.utils.FilterUtil;
  * EmptyLineSeparator
  * </li>
  * <li>
- * IllegalCatch
- * </li>
- * <li>
  * Indentation
  * </li>
  * <li>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java
@@ -59,7 +59,6 @@ public class XpathRegressionTest extends AbstractModuleTestSupport {
             "CommentsIndentation",
             "CustomImportOrder",
             "EmptyLineSeparator",
-            "IllegalCatch",
             "Indentation",
             "InterfaceIsType",
             "InterfaceMemberImpliedModifier",

--- a/src/xdocs/config_filters.xml
+++ b/src/xdocs/config_filters.xml
@@ -908,7 +908,6 @@ public class UserService {
           <li>CommentsIndentation</li>
           <li>CustomImportOrder</li>
           <li>EmptyLineSeparator</li>
-          <li>IllegalCatch</li>
           <li>Indentation</li>
           <li>InterfaceIsType</li>
           <li>InterfaceMemberImpliedModifier</li>

